### PR TITLE
Implement Invoices page with filters and opportunity to get an invoice

### PR DESCRIPTION
**Pull request by ilyawzrd**

### CODE REVIEW

File: client-app/src/pages/account/components/invoices-filter/invoices-filter.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    this.startDate = date;
+    this.dateChanged();
+  }
+
+  changeEndDate(date: Date) {
+    this.endDate = date;
+    this.dateChanged();
+  }
+
+  getDatepickerLocalization() {
```
Comment (t13ka) [inline client-app/src/pages/account/components/invoices-filter/invoices-filter.ts]: If you using the verb 'Get' then your function should return something. But you set values for datepicker labels then you have to use a verb 'Set' or something like that

File: client-app/src/pages/account/store/modules/invoices-list/index.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+
+// initial state
+export const initialState: InvoicesListState = {
+  errors: {},
+  invoicesList: {
+    listConfig: {
+      columns: invoicesGridFields,
+      pageNumber: 1,
+      pageSize: defaultPageSize,
```
Comment (t13ka) [inline client-app/src/pages/account/store/modules/invoices-list/index.ts]: Why are you using the defaultPageSize here but don't use the same for pageNumber? :)
It's a pretty infrequent scenario when we will have to change the defaultPageNumber, but what if we will?
Besides, it's better to don't use the magic numbers at all

**Comment by t13ka** [comment]
Doesn't work:
MicrosoftTeams-image
[errors4.txt](https://github.com/VirtoCommerce/vc-odt-mpa-theme/files/4352671/errors4.txt)

File: client-app/src/pages/account/components/invoices-filter/index.vue [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+          @show="toggleStatusDropdown()">
+          <template v-slot:button-content>
+            <div class="d-flex justify-content-between align-items-center">
+              {{ getCurrentStatusLabel() }}
+              <font-awesome-layers v-if="showStatusDropdown">
+                <font-awesome-icon :icon="faAngleUp" size="lg"></font-awesome-icon>
+              </font-awesome-layers>
+              <font-awesome-layers v-if="!showStatusDropdown">
+                <font-awesome-icon :icon="faAngleDown" size="lg"></font-awesome-icon>
+              </font-awesome-layers>
```
Comment (asvishnyakov) [inline client-app/src/pages/account/components/invoices-filter/index.vue]: As I can understand, bootstrap-vue's dropdown has angles by default. Why we need it?

File: client-app/src/pages/account/components/invoices-filter/index.vue [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    <div class="d-flex flex-wrap flex-sm-row flex-column">
+      <div class="col col-sm-6 col-md-5 col-lg-4">
+        <label for="begin-date">{{ $t("account.invoices.date-filter.from") }}</label>
+        <b-form-datepicker
+          id="begin-date"
+          :value="invoicesFilter.startDate"
+          value-as-date
+          reset-button
+          :label-reset-button="$t('account.invoices.date-filter.reset')"
+          :date-format-options="{ year: 'numeric', month: 'long', day: 'numeric' }"
```
Comment (asvishnyakov) [inline client-app/src/pages/account/components/invoices-filter/index.vue]: Seems like if you will just remove format optons, then everything will start working as expected:
https://bootstrap-vue.js.org/docs/components/form-datepicker#internationalization

File: client-app/src/pages/account/store/modules/invoices-list/actions.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  async [FETCH_INVOICES](context) {
+    context.commit(FETCH_INVOICES);
+    const searchCriteria = new PaymentSearchCriteria();
+    searchCriteria.pageNumber = context.state.invoicesList.listConfig.pageNumber;
+    searchCriteria.pageSize = context.state.invoicesList.listConfig.pageSize;
+    searchCriteria.sort = context.state.invoicesList.listConfig.filters.sort;
+    searchCriteria.statuses = context.state.invoicesList.listConfig.filters.statuses;
+    searchCriteria.startDate = context.state.invoicesList.listConfig.filters.startDate;
+    searchCriteria.endDate = context.state.invoicesList.listConfig.filters.endDate;
+    searchCriteria.keyword = context.state.invoicesList.listConfig.filters.keyword;
```
Comment (asvishnyakov) [inline client-app/src/pages/account/store/modules/invoices-list/actions.ts]: May it be replaced with something like
```typescript
searchCriteria =
{
  ...searchCriteria,
  ...context.state.invoicesList.listConfig.filters
}
```
?

File: client-app/src/pages/account/views/account-invoices/index.vue [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        striped
+        hover
+        :show-empty="true"
+        :empty-text="$t('account.invoices.no-invoices')"
+        :items="invoicesList.invoices"
+        :fields="invoicesList.listConfig.columns"
+        no-local-sorting
+        @sort-changed="sortChanged">
+        <template v-slot:cell(createdDate)="data">
+          <span>{{ data.value | moment("MMMM DD, YYYY") }}</span>
```
Comment (asvishnyakov) [inline client-app/src/pages/account/views/account-invoices/index.vue]: We probably need `L`, `LL` or somehow get current format from vue or bootstrap-vue internationalization settings
https://momentjs.com/docs/#/displaying/format/ Localized formats

File: client-app/src/pages/account/components/invoices-filter/index.vue (hunk 4 above)
Comment (ilyawzrd) [reply client-app/src/pages/account/components/invoices-filter/index.vue]: default date format is different
image

File: client-app/src/pages/account/components/invoices-filter/index.vue (hunk 3 above)
Comment (ilyawzrd) [reply client-app/src/pages/account/components/invoices-filter/index.vue]: by default, dropdown caret is too close to label. Therefore, for customization, I used FA with v-slot for button content
image

File: client-app/src/pages/account/views/account-invoices/index.vue (hunk 6 above)
Comment (ilyawzrd) [reply client-app/src/pages/account/views/account-invoices/index.vue]: done in last commit

File: client-app/src/pages/account/components/invoices-filter/index.vue (hunk 4 above)
Comment (asvishnyakov) [reply client-app/src/pages/account/components/invoices-filter/index.vue]: Default date format depends on your local date format

File: client-app/src/pages/account/store/modules/invoices-list/actions.ts (hunk 5 above)
Comment (ilyawzrd) [reply client-app/src/pages/account/store/modules/invoices-list/actions.ts]: I think it's worth leaving this option, it is readable and does not open up scope for an implicit error

File: client-app/src/pages/account/components/invoices-filter/index.vue (hunk 4 above)
Comment (ilyawzrd) [reply client-app/src/pages/account/components/invoices-filter/index.vue]: It depends, but default format for both EN and DE is too long

**Pull request merged**